src/main/java/org/irods/util/ResourceProperty.java

```diff
@@ -13,11 +20,21 @@ public enum ResourceProperty {
 
     private final String value;
 
-    ResourceProperty(String fieldName) {
-        this.value = fieldName;
+    /**
+     * Constructs a {@code ResourceProperty} enum constant with the specified string value.
+     *
+     * @param value The string representation of the user type.
+     */
+    ResourceProperty(String value) {
+        this.value = value;
     }
 
+    /**
+     * Returns the string representation of this property.
+     *
+     * @return The string representation of the property.
```

Comment:
Probably don't need the bottom two lines because the main description covers it.

Reply:
Done, I'll do the same for the other enums.